pkg/reconciler/taskrun/taskrun.go

```diff
@@ -386,6 +380,13 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1beta1.TaskRun, rtr *re
 	var pod *corev1.Pod
 	var err error
 
+	if tr.ObjectMeta.Labels == nil {
```

Comment:
TODO: these nil checks used to be higher in the `prepare` func (lines 297 and 312). Put them back to roughly the same spot.